os/net/ipv6/uipbuf.h

```diff
@@ -35,33 +35,98 @@
+ * \param protocol A pointer to a variable where the protocol of the header will be stored
+ * \retval         returns address of the starting position of the next header
+ *
+ *                 This function moves to the last header of the IPv6 packet.
+ */
 uint8_t* uipbuf_get_last_header(uint8_t *buffer, uint16_t size, uint8_t *protocol);
 
-/* Attributes relating to the current packet in uipbuf */
+
+/* \brief          Get the value of the attribute
+ * \param type     The attribute to get the value of
+ * \retval         the value of the attribute
+ *
+ *                 This function gets the value of a specific uipbuf attribute.
+ */
 uint16_t uipbuf_get_attr(uint8_t type);
-void uipbuf_set_attr_flag(uint16_t flag);
-void uipbuf_clr_attr_flag(uint16_t flag);
-uint16_t uipbuf_is_attr_flag(uint16_t flag);
+
+
+/* \brief          Set the value of the attribute
+ * \param type     The attribute to set the value of
+ * \param value    The value to set
+ * \retval         0 - indicates failure of setting the value
+ * \retval         1 - indicates success of setting the value
+ *
+ *                 This function sets the value of a specific uipbuf attribute.
+ */
 int uipbuf_set_attr(uint8_t type, uint16_t value);
+
+/* \brief          Set bits in the uipbuf attribute flags.
+ * \param flag_bits The bits to set in the flag.
+ *
+ *                 This function sets the uipbuf attributes flag of specified bits.
+ */
+void uipbuf_set_attr_flag(uint16_t flag_bits);
+
+/* \brief          Clear bits in the uipbuf attribute flags.
+ * \param flag_bits The bits to clear in the flag.
+ *
+ *                 This function clears the uipbuf attributes flag of specified bits.
+ */
+void uipbuf_clr_attr_flag(uint16_t flag_bits);
+
+/* \brief          Check if bits in the uipbuf attribute flag are set.
+ * \param flag_bits The bits to check in the flag.
+ *
+ *                 This function checks if the specified bits are set in the
+ *                 uipbuf attributes flag.
+ */
+uint16_t uipbuf_is_attr_flag(uint16_t flag_bits);
+
+
+/* \brief          Clear all attributes.
+ *
+ *                 This function clear all attributes in the uipbuf attributes
+ *                 including all flags.
+ */
 void uipbuf_clear_attr(void);
 
-/* These flags will be used for being */
+/**
+ * \brief The bits defined for uipbuf attributes flag.
+ *
+ */
+/* Avoid using NHC compression on the packet (6LoWPAN) */
 #define UIPBUF_ATTR_FLAGS_6LOWPAN_NO_NHC_COMPRESSION      0x01
+/* Avoid using prefix compression on the packet (6LoWPAN) */
 #define UIPBUF_ATTR_FLAGS_6LOWPAN_NO_PREFIX_COMPRESSION   0x02
 
+/**
+ * \brief The attributes defined for uipbuf attributes function.
+ *
+ */
 enum {
-  UIPBUF_ATTR_LLSEC_LEVEL,
-  UIPBUF_ATTR_LLSEC_KEY_ID,
-  UIPBUF_ATTR_INTERFACE_ID,
-  UIPBUF_ATTR_PHYSICAL_NETWORK_ID,
-  UIPBUF_ATTR_MAX_MAC_TRANSMISSIONS,
-  UIPBUF_ATTR_FLAGS,
+  UIPBUF_ATTR_LLSEC_LEVEL,      /* Control link layer security level. */
```

Comment:
I think (didn't test) you need `/**<` here for doxygen to generate HTML

Reply:
Maybe I should try adding that then!